includes/admin/extensions/class-extension-manager.php

```diff
@@ -0,0 +1,484 @@
+	 */
+	public $all_plugins;
+
+	/**
+	 * The minimum pass ID required to install the extension.
+	 *
+	 * @since 2.11.x
+	 * @var int
+	 */
+	private $required_pass_id;
+
+	/**
+	 * Pass Manager class
+	 *
+	 * @var \EDD\Admin\Pass_Manager
+	 */
+	protected $pass_manager;
+
+	public function __construct( $required_pass_id = null ) {
+		if ( $required_pass_id ) {
+			$this->required_pass_id = $required_pass_id;
+		}
+		$this->pass_manager = new Pass_Manager();
+
+		add_action( 'wp_ajax_edd_activate_extension', array( $this, 'activate' ) );
+		add_action( 'wp_ajax_edd_install_extension', array( $this, 'install' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'register_scripts' ) );
+		// add_action( 'edd_plugin_activated', array( $this, 'delete_options' ) );
+	}
+
+	/**
+	 * Registers the extension manager script.
+	 *
+	 * @since 2.11.x
+	 * @return void
+	 */
+	public function register_scripts() {
+		$minify = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ? '' : '.min';
+		wp_register_script( 'edd-extension-manager', EDD_PLUGIN_URL . "assets/js/extension-manager{$minify}.js", array( 'jquery' ), EDD_VERSION, true );
+		wp_localize_script(
+			'edd-extension-manager',
+			'EDDExtensionManager',
+			array(
+				'activating'               => __( 'Activating', 'easy-digital-downloads' ),
+				'installing'               => __( 'Installing', 'easy-digital-downloads' ),
+				'plugin_install_failed'    => __( 'Could not install the plugin. Please download and install it manually via Plugins > Add New.', 'easy-digital-downloads' ),
+				'extension_install_failed' => sprintf(
+					/* translators: 1. opening anchor tag, do not translate; 2. closing anchor tag, do not translate */
+					__( 'Could not install the extension. Please %1$sdownload it from your account%2$s and install it manually.', 'easy-digital-downloads' ),
+					'<a href="https://easydigitaldownloads.com/your-account/" target="_blank" rel="noopener noreferrer">',
+					'</a>'
+				),
+				'extension_manager_nonce'  => wp_create_nonce( 'edd_extensionmanager' ),
+			)
+		);
+	}
+
+	/**
+	 * Outputs a standard extension card.
+	 *
+	 * @since 2.11.x
+	 * @param array  $product             The extension information retrieved from the Products API.
+	 * @param array  $inactive_parameters The array of information to build the button for an inactive/not installed plugin.
+	 * @param array  $active_parameters   The array of information needed to build the link to configure an active plugin.
+	 * @return void
+	 */
+	public function do_extension_card( $product, $inactive_parameters, $active_parameters ) {
+		if ( ! $product ) {
+			return;
+		}
+		$base_class   = 'edd-extension-manager__card';
+		$card_classes = array(
+			$base_class,
+		);
+		$variation    = 'stacked';
+		if ( ! empty( $product['card'] ) ) {
+			$variation = $product['card'];
+		}
+		$card_classes[] = "{$base_class}--{$variation}";
+		?>
+		<div class="<?php echo esc_attr( implode( ' ', $card_classes ) ); ?>">
```

Comment:
```suggestion
		<div class="<?php echo esc_attr( implode( ' ', array_map( 'sanitize_html_class', $card_classes ) ) ); ?>">
```

I like the sanitize function here because the source of the class names isn't really clear.